core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java

```diff
@@ -84,6 +82,38 @@ public DataMapWriter getDataMapWriter(AbsoluteTableIdentifier identifier, String
     }
   }
 
+  private CarbonFile[] getCarbonIndexFiles(String segmentId) {
+    String path = identifier.getTablePath() + "/Fact/Part0/Segment_" + segmentId;
+    FileFactory.FileType fileType = FileFactory.getFileType(path);
+    CarbonFile carbonFile = FileFactory.getCarbonFile(path, fileType);
```

Comment:
use `public static CarbonFile getCarbonFile(String path, FileType fileType)`

Reply:
You mean use `FileFactory.getCarbonFile(path)` ? I have changed